### PR TITLE
test: add unit tests for p2p downloader

### PR DESCRIPTION
**Pull request by Starnop**
Signed-off-by: Starnop <starnop@163.com>

### Ⅰ. Describe what this PR did
As title described, this PR add partial unit tests for the p2p downloader module.

### Ⅱ. Does this pull request fix one issue?

fixes part of #352 

### Ⅲ. Why don't you add test cases (unit test/integration test)? (你真的觉得不需要加测试吗？)
exactly

### CODE REVIEW

**Comment by lowzj** [comment]
Add the following code into `back_downloader_test.go` to test `BackDownloader`: 
```go
func Test(t *testing.T) {
	check.TestingT(t)
}
```

File: dfget/core/downloader/p2p_downloader/client_writer.go [hunk 1]
```diff
@@ -140,11 +140,13 @@ func (cw *ClientWriter) Run() {
...
 			continue
 		}
-		if err := cw.write(piece, time.Now()); err != nil {
+		startTime := time.Now()
+		if err := cw.write(piece); err != nil {
 			logrus.Errorf("write item:%s error:%v", piece, err)
 			cw.cfg.BackSourceReason = config.BackSourceReasonWriteError
 			cw.result = false
 		}
+		go cw.sendSuccessPiece(piece, time.Since(startTime))
```
[the remark this answers is not in the document]
Comment (Starnop) [reply dfget/core/downloader/p2p_downloader/client_writer.go]: Done.

**Comment by Starnop** [comment]
@lowzj Updated. 😄

**Comment by lowzj** [comment]
LGTM

**Pull request merged**